src/util/transaction_identifier.h

```diff
@@ -0,0 +1,67 @@
+#ifndef BITCOIN_UTIL_TRANSACTION_IDENTIFIER_H
+#define BITCOIN_UTIL_TRANSACTION_IDENTIFIER_H
+
+#include <uint256.h>
+#include <util/types.h>
+
+/** transaction_identifier represents the two canonical transaction identifier
+ * types (txid, wtxid).*/
+template <bool has_witness>
+class transaction_identifier
+{
+    uint256 m_wrapped;
+
+    // Note: Use FromUint256 externally instead.
+    transaction_identifier(const uint256& wrapped) : m_wrapped{wrapped} {}
```

Comment:
That might be better but would allow something like `Txid{Wtxid{}}` right now (because of the conversion function). We could make it a template like we do for `Compare`...

Reply:
But (a) that would presumably be silly to write, and (b) the conversion function is going away, which will ensure any code like that also goes away.